### PR TITLE
Fix type of Bernoulli sample type

### PR DESCRIPTION
**Pull request by juliohm**
Includes #1046 . I messed up with Git again, so creating a new PR from scratch.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -55,6 +55,7 @@ const tests = [
     "chernoff",
     "univariate_bounds",
     "negativebinomial",
+    "bernoulli"
```
Comment (matbesancon) [inline test/runtests.jl]:
```suggestion
    "bernoulli",
```

**Comment by matbesancon** [comment]
Thanks!

**Comment by MikeInnes** [comment]
Just got bitten by this change (which probably should have been treated as breaking).

This seems like a very strange decision to me. `Bool` is useful because it's the narrowest type that represents what `Bernoulli` can output (i.e. `0` or `1`). It's additionally useful because it can be used as a condition, which is pretty frequent in probabilistic models; I'd rather not write `Bool(rand(Bernoulli(p)))` or `rand(Bernoulli(p))) == 1` every time there. (There's no similar downside to emitting `Bool` since that can always be casted to a richer type automatically.)

I'd strongly be in favour of reverting this (the quantum processors can likely look after themselves.)

**Comment by matbesancon** [comment]
Indeed I made the mistake of tagging post-merge as a patch release.

As for the preferred type, as mentioned I had no strong opinion for either,
Bool being more handy but using it as Int is still a very Julian thing.

I'll let Julio give his opinion, but would be ok with switching to Bool, as
long as we maintain consistency (which was broken before this PR)

**Comment by matbesancon** [comment]
cc @juliohm

**Comment by juliohm** [comment]
Sorry for the late reply, just saw this in my inbox today.

It seems strange to me to rely on a computer science type to define
Bernoulli variables. Can you please confirm that performing operations with
large Boolean arrays representing samples of various Bernoulli does not
overflow nor slows down computations in practice? Part of my concern is
that we want to do math with the 0s and 1s in this context. It is not
Boolean semantic that we are doing here. In most cases where Bernoulli is
being used as a check rand(Bernoulli(p)) == 1,  one could actually write
the much simpler rand() < p. I still think that using Bool here is strange,
my vote is in favor of Int.

**Comment by matbesancon** [comment]
> samples of various Bernoulli does not overflow nor slows down computations in practice

any arithmetic on booleans converts them to Int, so no overflow: `true + true == 2`

**Comment by juliohm** [comment]
So this means conversions take place? We loose performance then correct ?

**Comment by matbesancon** [comment]
I don't think the cost here is awful, if there is any

**Pull request merged**